src/lib/operators/join_sort_merge/radix_cluster_sort.hpp

```diff
@@ -283,19 +283,18 @@ class RadixClusterSort {
     const std::vector<T> split_values = _pick_split_values(sample_values);
 
     // Implements range clustering
-    auto cluster_count = _cluster_count;
-    auto clusterer = [cluster_count, &split_values](const T& value) {
+    auto clusterer = [&split_values](const T& value) {
       // Find the first split value that is greater or equal to the entry.
       // The split values are sorted in ascending order.
       // Note: can we do this faster? (binary search?)
-      for (size_t cluster_id = 0; cluster_id < cluster_count - 1; ++cluster_id) {
+      for (size_t cluster_id = 0; cluster_id < split_values.size(); ++cluster_id) {
```

Comment:
Why is `split_values.size() != _cluster_count`?

Reply:
Split values are the values at the splits of the array. So splits for `[1,3,5]` could be `[2,4]`.

Reply:
I'll add a comment for the `split_values` array.

Reply:
And  why is `-1` incorrect then?

Reply:
Because we might have less split values than clusters. E.g. for a column with few distinct values.

Reply:
Do the new comments help?